addons/website_cf_turnstile/static/src/js/turnstile.js

```diff
@@ -4,52 +4,116 @@ import "@website/snippets/s_website_form/000";  // force deps
-                const mode = new URLSearchParams(window.location.search).get('cf') == 'show' ? 'always' : 'interaction-only';
-                $(`<div class="s_turnstile cf-turnstile float-end"
-                         data-action="website_form"
-                         data-appearance="${mode}"
-                         data-response-field-name="turnstile_captcha"
-                         data-sitekey="${session.turnstile_site_key}"
-                         data-error-callback="throwTurnstileError"
-                    ></div>
-                    <script class="s_turnstile">
-                        // Rethrow the error, or we only will catch a "Script error" without any info 
-                        // because of the script api.js originating from a different domain.
-                        function throwTurnstileError(code) {
-                            const error = new Error("Turnstile Error");
-                            error.code = code;
-                            throw error;
+export const turnStile = {
+    addTurnstile: function (action) {
+        if (!this.isEditable && !this.$('.s_turnstile').length) {
+            const mode = new URLSearchParams(window.location.search).get('cf') == 'show' ? 'always' : 'interaction-only';
+            return $(`<div class="s_turnstile cf-turnstile float-end"
+                        style="display: none;"
+                        data-action="${action}"
+                        data-appearance="${mode}"
+                        data-response-field-name="turnstile_captcha"
+                        data-sitekey="${session.turnstile_site_key}"
+                        data-error-callback="throwTurnstileError"
+                        data-callback="turnstileSuccess"
+                        data-before-interactive-callback="turnstileBecomeVisible"
+                ></div>
+                <script class="s_turnstile">
+                    // Rethrow the error, or we only will catch a "Script error" without any info
+                    // because of the script api.js originating from a different domain.
+                    function throwTurnstileError(code) {
+                        const error = new Error("Turnstile Error");
+                        error.code = code;
+                        throw error;
+                    }
+                    function turnstileSuccess() {
+                        let form = this.wrapper.parentElement.parentElement;
+                        let formButton = form.querySelector("button[type='submit']");
+                        let webButton  = form.querySelector("a.s_website_form_send[role='button']");
+                        if (formButton) {
+                            formButton.disabled = false;
+                            formButton.classList.remove('disabled');
+                            formButton.querySelector('i.turnstile-spinner').remove();
```

Comment:
I think there is a way to have a common code for both cases website form / reset password
```js
                    function turnstileSuccess() {
                        const form = this.wrapper.parentElement.parentElement;
                        const spinner = form.querySelector("i.turnstile-spinner");
                        const button = spinner.parentElement;
                        button.disabled = false;
                        button.classList.remove('disabled');
                        spinner.remove();
                    }
```